gqcp/tests/Operator/SecondQuantized/SQOneElectronOperator_test.cpp

```diff
@@ -238,3 +285,112 @@ BOOST_AUTO_TEST_CASE ( calculateExpectationValue_throw ) {
     BOOST_CHECK_THROW(h.calculateExpectationValue(D_invalid), std::invalid_argument);
     BOOST_CHECK_NO_THROW(h.calculateExpectationValue(D_valid));
 }
+
+
+/**
+ * Check whether or not calculateExpectationValue shows the correct behaviour
+ */
+BOOST_AUTO_TEST_CASE ( calculateExpectationValue_behaviour ) {
+    
+    const size_t dim = 2;
+
+    // Initialize a test matrix and convert it into an operator
+    GQCP::QCMatrix<double> M1 (dim);
+    M1 << 1.0, 2.0,
+          3.0, 4.0;
+    const GQCP::ScalarSQOneElectronOperator<double> op (M1);
+
+    // initialize an alpha and beta density matrix, each one is chosen as a hermitian matrix.
+    GQCP::QCMatrix<double> d (dim);
+    d << 0.0, 1.0,
+         1.0, 0.0;
+    
+    // Initialize a reference value
+    GQCP::QCMatrix<double> ref (1);
+    ref << 5.0;
+
+    const auto ex_value = op.calculateExpectationValue(d);
+    BOOST_CHECK(ex_value.isApprox(ref, 1.0e-08));
```

Comment:
```suggestion
    GQCP::QCMatrix<double> D (dim);
    D << 0.0, 1.0,
         1.0, 0.0;
    
    // Initialize a reference value and check the result.
    const double reference_expectation_value = 5.0;

    const auto expectation_value = op.calculateExpectationValue(D)(0);  // extract the 'scalar' from a one-dimensional vector
    BOOST_CHECK(expectation_value.isApprox(reference_expectation_value, 1.0e-08));
```

The 1-DM is usually given the symbol `D` instead of `d`.